src/components/frame/TokenActivation/TokensTab/ChangeTokenStateForm.tsx

```diff
@@ -0,0 +1,263 @@
+
+const displayName = 'frame.TokenActivation.TokensTab.ChangeTokenStateForm';
+
+const MSG = defineMessages({
+  tokenActivation: {
+    id: `${displayName}.tokenActivation`,
+    defaultMessage: 'Token activation',
+  },
+  activate: {
+    id: `${displayName}.activate`,
+    defaultMessage: 'Activate',
+  },
+  deactivate: {
+    id: `${displayName}.deactivate`,
+    defaultMessage: 'Deactivate',
+  },
+  balance: {
+    id: `${displayName}.balance`,
+    defaultMessage: 'balance: {tokenBalance}',
+  },
+  locked: {
+    id: `${displayName}.locked`,
+    defaultMessage: 'Locked: {lockedTokens}',
+  },
+  lockedTooltip: {
+    id: `${displayName}.lockedTooltip`,
+    defaultMessage: `You have unclaimed transactions which must be claimed
+    before these tokens can be withdrawn.`,
+  },
+  max: {
+    id: `${displayName}.max`,
+    defaultMessage: 'Max',
+  },
+});
+
+const validationSchema = object({
+  amount: number()
+    .transform((value) => toFinite(value))
+    .required()
+    .moreThan(0),
+}).defined();
+
+type FormValues = InferType<typeof validationSchema>;
+
+export interface ChangeTokenStateFormProps {
+  token: Token;
+  tokenDecimals: number;
+  activeTokens: BigNumber;
+  inactiveTokens: BigNumber;
+  lockedTokens: BigNumber;
+  hasLockedTokens: boolean;
+  colonyAddress: Address;
+}
+
+const ChangeTokenStateForm = ({
+  token,
+  tokenDecimals,
+  activeTokens,
+  inactiveTokens,
+  lockedTokens,
+  hasLockedTokens,
+  colonyAddress,
+}: ChangeTokenStateFormProps) => {
+  const [isActivate, setIsActive] = useState(true);
+
+  const { wallet } = useAppContext();
+
+  const formattedActiveTokens = getFormattedTokenValue(
+    activeTokens,
+    token.decimals,
+  );
+  const formattedInactiveTokens = getFormattedTokenValue(
+    inactiveTokens,
+    token.decimals,
+  );
+  const formattedLockedTokens = getFormattedTokenValue(
+    lockedTokens,
+    token.decimals,
+  );
+  const unformattedTokenBalance = moveDecimal(
+    isActivate ? inactiveTokens : activeTokens,
+    -tokenDecimals,
+  );
+
+  const tokenBalance = useMemo(
+    () => (isActivate ? formattedInactiveTokens : formattedActiveTokens),
+    [isActivate, formattedActiveTokens, formattedInactiveTokens],
+  );
+
+  const formAction = useCallback(
+    (actionType: '' | '_ERROR' | '_SUCCESS') =>
+      isActivate
+        ? ActionTypes[`USER_DEPOSIT_TOKEN${actionType}`]
+        : ActionTypes[`USER_WITHDRAW_TOKEN${actionType}`],
+    [isActivate],
+  );
+
+  const transform = pipe(
+    mapPayload(({ amount }) => {
+      // Convert amount string with decimals to BigInt (eth to wei)
+      const formattedAmount = BigNumber.from(
+        moveDecimal(amount, tokenDecimals),
+      );
+
+      return {
+        amount: formattedAmount,
+        userAddress: wallet?.address,
+        colonyAddress,
+        tokenAddress: token.tokenAddress,
+      };
+    }),
+  );
+
+  const handleSubmitSuccess = useCallback((res, values, { reset }) => {
```

Comment:
I don't think having these `useMemo` and `useCallback` here will make a difference as they are memorizing things that should be simple to calculate